src/simple_ip_input.rs

```diff
@@ -0,0 +1,155 @@
+use crate::{input::view_asterisk, language::Language, text, Texts};
+use json_gettext::get_text;
+use std::cell::RefCell;
+use std::rc::Rc;
+use wasm_bindgen::JsCast;
+use web_sys::HtmlInputElement;
+use yew::{html, Component, Context, Html, InputEvent, Properties};
+const DEFAULT_MAX_HEIGHT: u32 = 280;
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct Ip {
+    pub start: Option<String>,
+    pub end: Option<String>,
+}
+
+#[derive(Clone, PartialEq, Eq)]
+pub enum Message {
+    InputString(String),
+    InputError,
+}
+
+pub struct Model {
+    input: String,
+}
+
+#[derive(Clone, Eq, PartialEq, Properties)]
+pub struct Props {
+    pub txt: Texts,
+    pub language: Language,
+    #[prop_or(None)]
+    pub width: Option<u32>,
+    #[prop_or(DEFAULT_MAX_HEIGHT)]
+    pub max_height: u32,
+    pub input_data: Rc<RefCell<Option<Ip>>>,
+    #[prop_or(None)]
+    pub title: Option<String>,
+    #[prop_or(None)]
+    pub example_message: Option<String>,
+    #[prop_or(None)]
+    pub placeholder_message: Option<String>,
+    #[prop_or(false)]
+    pub required: bool,
+}
+
+impl Component for Model {
+    type Message = Message;
+    type Properties = Props;
+
+    fn create(_ctx: &Context<Self>) -> Self {
+        Self {
+            input: String::new(),
+        }
+    }
+
+    fn changed(&mut self, _ctx: &Context<Self>, _old_props: &Self::Properties) -> bool {
+        true
+    }
+
+    fn update(&mut self, ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            Message::InputString(text) => {
+                self.input = text.clone();
+                ctx.props()
+                    .input_data
+                    .try_borrow_mut()
+                    .ok()
+                    .map(|mut data| {
+                        if text.contains('~') {
+                            text.split_once('~').and_then(|(start, end)| {
+                                data.replace(Ip {
+                                    start: Some(start.to_string()),
+                                    end: Some(end.to_string()),
+                                })
+                            })
+                        } else {
+                            data.replace(Ip {
+                                start: Some(text.trim().to_string()),
+                                end: None,
+                            })
+                        }
+                    });
+            }
```

Comment:
I don't find any code which validates user input. Wrong input should be handled properly like by showing good messages and erasing the wrong input.

Reply:
![Uploading Screenshot 2023-05-15 at 1.55.05 PM.png…]()

Reply:
<img width="1920" alt="Screenshot 2023-05-15 at 1 55 05 PM" src="https://github.com/aicers/frontary/assets/102562226/8e7c5c39-6c1a-47ad-87c1-586d4d56ce21">

Done